### PR TITLE
Allow routing same K8s manifest to multiple platform providers

### PR DESCRIPTION
**Pull request by nghialv**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/app/piped/executor/kubernetes/applier_group.go [hunk 1]
```diff
@@ -46,48 +49,99 @@ func newApplierGroup(defaultProvider string, appCfg config.KubernetesApplication
...
+				return nil, fmt.Errorf("provider %s specified in resourceRoutes was not found", name)
+			}
+			d.appliers[name] = provider.NewApplier(appCfg.Input, *cp.KubernetesConfig, logger)
 			continue
 		}
-
-		cp, ok := pipedCfg.FindPlatformProvider(providerName, model.ApplicationKind_KUBERNETES)
-		if !ok {
-			return nil, fmt.Errorf("provider %s specified in resourceRoutes was not found", providerName)
+		if labels := r.Provider.Labels; len(labels) > 0 {
```
Comment (knanao) [inline pkg/app/piped/executor/kubernetes/applier_group.go:70]: Sorry, Let me check one point.
Is it possible for a user to identify a specific platform provider without that name?

**Review by knanao — DISMISSED** [review]

File: pkg/app/piped/executor/kubernetes/applier_group.go (hunk 1 above)
Comment (nghialv) [reply pkg/app/piped/executor/kubernetes/applier_group.go:70]: Yes, I assume some use cases as below:
- using only name to specify exactly one provider
- using labels without name to specify multiple providers

So it is possible to give only the labels without name.
Wdyt?

Comment (knanao) [reply pkg/app/piped/executor/kubernetes/applier_group.go:70]: That sounds good! Thank you. 🙌

File: pkg/config/piped.go [hunk 2]
```diff
@@ -218,6 +218,32 @@ func (s *PipedSpec) FindPlatformProvider(name string, t model.ApplicationKind) (
 	return PipedPlatformProvider{}, false
 }
 
+// FindPlatformProviderByLabel finds all PlatformProviders which match the provided labels.
+func (s *PipedSpec) FindPlatformProviderByLabel(labels map[string]string, t model.ApplicationKind) []PipedPlatformProvider {
```
Comment (khanhtc1202) [inline pkg/config/piped.go:222]:
```suggestion
// FindPlatformProvidersByLabel finds all PlatformProviders which match the provided labels.
func (s *PipedSpec) FindPlatformProvidersByLabel(labels map[string]string, t model.ApplicationKind) []PipedPlatformProvider {
```

Comment (nghialv) [reply pkg/config/piped.go:222]: Nice.

File: pkg/app/piped/executor/kubernetes/applier_group.go [hunk 3]
```diff
@@ -46,48 +49,99 @@ func newApplierGroup(defaultProvider string, appCfg config.KubernetesApplication
...
+			cps := pipedCfg.FindPlatformProviderByLabel(labels, model.ApplicationKind_KUBERNETES)
```
Comment (nghialv) [inline pkg/app/piped/executor/kubernetes/applier_group.go:71]:
```suggestion
			cps := pipedCfg.FindPlatformProvidersByLabel(labels, model.ApplicationKind_KUBERNETES)
```

File: pkg/config/piped_test.go [hunk 4]
```diff
@@ -1134,3 +1134,95 @@ func TestPipedSpecClone(t *testing.T) {
 		})
 	}
 }
+
+func TestFindPlatformProviderByLabel(t *testing.T) {
```
Comment (nghialv) [inline pkg/config/piped_test.go:1138]:
```suggestion
func TestFindPlatformProvidersByLabel(t *testing.T) {
```

File: pkg/config/piped_test.go [hunk 5]
```diff
@@ -1134,3 +1134,95 @@ func TestPipedSpecClone(t *testing.T) {
...
+						"foo":   "foo-1",
+					},
+				},
+			},
+		},
+	}
+
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			got := pipedSpec.FindPlatformProviderByLabel(tc.labels, model.ApplicationKind_KUBERNETES)
```
Comment (nghialv) [inline pkg/config/piped_test.go:1224]:
```suggestion
			got := pipedSpec.FindPlatformProvidersByLabel(tc.labels, model.ApplicationKind_KUBERNETES)
```

**Review by knanao — APPROVED** [review]

**Review by khanhtc1202 — APPROVED** [review]
Nice 👍

**Pull request merged**